sdk/storage/Azure.Storage.Common/src/Shared/StorageClientOptions.cs

```diff
@@ -64,21 +59,23 @@ public static HttpPipelinePolicy AsPolicy<TUriBuilder>(this AzureSasCredential c
         /// Get an authentication policy to sign Storage requests.
         /// </summary>
         /// <param name="credential">Credential to use.</param>
+        /// <param name="scope">Scope to use.</param>
         /// <param name="options"> The <see cref="ISupportsTenantIdChallenges"/> to apply to the credential. </param>
         /// <returns>An authentication policy.</returns>
-        public static HttpPipelinePolicy AsPolicy(this TokenCredential credential, ClientOptions options) =>
+        public static HttpPipelinePolicy AsPolicy(this TokenCredential credential, string scope, ClientOptions options) =>
             new StorageBearerTokenChallengeAuthorizationPolicy(
                 credential ?? throw Errors.ArgumentNull(nameof(credential)),
-                StorageScope,
+                scope,
```

Comment:
Looks like this doesn't handle scope being null, which it seems it will be by default based off of the callsite below. Should this be `scope ?? StorageScope` instead?

Reply:
Done